Content.Client/GameObjects/EntitySystems/GhostSystem.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Content.Client.GameObjects.Components.Observer;
+using Content.Client.UserInterface;
+using Content.Shared.GameObjects.EntitySystems;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Client.Player;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+
+namespace Content.Client.GameObjects.EntitySystems
+{
+    [UsedImplicitly]
+    public class GhostSystem : SharedGhostSystem
+    {
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+        [Dependency] private readonly IGameHud _gameHud = default!;
+
+        private bool _ghostVisibility;
+
+        private bool GhostVisibility
+        {
+            get => _ghostVisibility;
+            set
+            {
+                if (_ghostVisibility == value)
+                {
+                    return;
+                }
+
+                _ghostVisibility = value;
+
+                foreach (var ghost in ComponentManager.GetAllComponents(typeof(GhostComponent), true))
+                {
+                    if (ghost.Owner.TryGetComponent(out SpriteComponent? sprite))
+                    {
+                        sprite.Visible = value;
+                    }
+                }
+            }
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<GhostComponent, ComponentInit>(OnGhostInit);
+            SubscribeLocalEvent<GhostComponent, ComponentRemove>(OnGhostRemove);
+
+            SubscribeLocalEvent<GhostComponent, PlayerAttachedEvent>(OnGhostPlayerAttach);
+            SubscribeLocalEvent<GhostComponent, PlayerDetachedEvent>(OnGhostPlayerDetach);
+
+            SubscribeNetworkEvent<GhostWarpsResponseEvent>(OnGhostWarpsResponse);
+        }
+
+        private void OnGhostInit(EntityUid uid, GhostComponent component, ComponentInit args)
+        {
+            if (component.Owner.TryGetComponent(out SpriteComponent? sprite))
+            {
+                sprite.Visible = _playerManager
```

Comment:
Make this set it to `GhostVisibility` instead of this long check, maybe?

Reply:
Done in https://github.com/space-wizards/space-station-14/pull/4159/commits/d1d4589647b7c53883d0ac2fcf4b58d7bf5f7131